org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/LinkResourceUtil.java

```diff
@@ -0,0 +1,127 @@
+
+	private static boolean isNewer(Path file, Instant instant) throws CoreException {
+		try {
+			BasicFileAttributes attributes = Files.getFileAttributeView(file, BasicFileAttributeView.class).readAttributes();
+			return attributes.creationTime().toInstant().isAfter(instant);
+		} catch (IOException ex) {
+			throw new CoreException(Status.error(ex.getMessage(), ex));
+		}
+	}
+
+    /**
+     * Get the redirected path of the input path. The path will be redirected to
+     * the workspace's metadata folder ({@link LinkResourceUtil#METADATA_FOLDER_PATH}).
+     * @param projectName name of the project.
+     * @param path path needs to be redirected.
+     * @return the redirected path.
+     */
+    private static IPath getMetaDataFilePath(String projectName, IPath path) {
+        if (path.segmentCount() == 1) {
+            return METADATA_FOLDER_PATH.append(projectName).append(path);
+        }
+
+        String lastSegment = path.lastSegment();
+        if (IProjectDescription.DESCRIPTION_FILE_NAME.equals(lastSegment)) {
+            return METADATA_FOLDER_PATH.append(projectName).append(lastSegment);
+        }
+
+        return null;
+    }
+
+	private static void linkFolderIfNewer(IFolder settingsFolder, Instant instant) throws CoreException {
+		if (settingsFolder.isLinked()) {
+			return;
+		}
+		if (settingsFolder.exists() && !isNewer(settingsFolder.getLocation().toPath(), instant)) {
+			return;
+		}
+		if (!settingsFolder.exists()) {
+			// not existing yet, create link
+			File diskFolder = getMetaDataFilePath(settingsFolder.getProject().getName(), settingsFolder.getProjectRelativePath()).toFile();
+			diskFolder.mkdirs();
+			settingsFolder.createLink(diskFolder.toURI(), IResource.NONE, new NullProgressMonitor());
+		} else if (isNewer(settingsFolder.getLocation().toPath(), instant)) {
+			// already existing but not existing before import: move then link
+			File sourceFolder = settingsFolder.getLocation().toFile();
+			File targetFolder = getMetaDataFilePath(settingsFolder.getProject().getName(), settingsFolder.getProjectRelativePath()).toFile();
+			File parentTargetFolder = targetFolder.getParentFile();
+			if (!parentTargetFolder.isDirectory()) {
+				parentTargetFolder.mkdirs();
+			}
+			sourceFolder.renameTo(targetFolder);
+			settingsFolder.createLink(targetFolder.toURI(), IResource.REPLACE, new NullProgressMonitor());
+		}
+	}
+
+	private static void linkFileIfNewer(IFile metadataFile, Instant instant, String ifEmpty) throws CoreException {
+		if (metadataFile.isLinked()) {
+			return;
+		}
+		if (metadataFile.exists() && !isNewer(metadataFile.getLocation().toPath(), instant)) {
+			return;
+		}
+		File targetFile = getMetaDataFilePath(metadataFile.getProject().getName(), metadataFile.getProjectRelativePath()).toFile();
+		if (!targetFile.exists()) {
+			try {
+				targetFile.getParentFile().mkdirs();
+				targetFile.createNewFile();
```

Comment:
Since we are using `StandardCopyOption.REPLACE_EXISTING`, I think `createNewFile()` can be removed?